### PR TITLE
BUG: Various bug-fixes and enhancements from testing over last few weeks.

### PR DESCRIPTION
**Pull request by tavaughan**

### CODE REVIEW

File: LumpNav/LumpNav.py [hunk 1]
```diff
@@ -452,8 +454,9 @@ def disconnect(self):#TODO see connect
     self.deleteAllFiducialsButton.disconnect('clicked()', self.onDeleteAllFiducialsClicked)
     self.placeButton.disconnect('clicked(bool)', self.onPlaceClicked)
 
-    self.rightCameraButton.disconnect('clicked()', self.onRightCameraButtonClicked)
-    self.leftCameraButton.disconnect('clicked()', self.onLeftCameraButtonClicked)
+    self.rightBullseyeCameraButton.disconnect('clicked()', self.onRightCameraButtonClicked)
```
Comment (lassoan) [inline LumpNav/LumpNav.py]: This disconnect does not match the connect.

File: LumpNav/LumpNav.py [hunk 2]
```diff
@@ -843,101 +856,135 @@ def getViewNode(self, viewName):
...
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].followSetSafeYMaximum(0.6)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].followSetModelNode(self.tumorModel_Needle)
-      self.viewpointLogic.nodeInstanceDictionary[viewNode].followStart()
+  def updateGUIButtons(self):
+    logging.debug("updateGUIButtons")
+    
+    leftViewNode = self.getViewNode('View1')
+    
+    if (self.viewpointLogic.getViewpointForViewNode(leftViewNode).isCurrentModeAutoCenter()):
+      self.leftAutoCenterCameraButton.setChecked(True)
```
Comment (lassoan) [inline LumpNav/LumpNav.py]: Need to call blockSignals to make sure the button state change does not trigger a MRML update.

**Comment by lassoan** [comment]
Thank you Thomas. Please fix the two small comments in a separate pull request.

File: LumpNav/LumpNav.py (hunk 2 above)
Comment (tavaughan) [inline LumpNav/LumpNav.py]: Sorry, I'm not sure what this means. Are there code examples I can look at?

Comment (lassoan) [inline LumpNav/LumpNav.py]: blockSignals prevents the widget to emit signals when you change it. In this case, probably you want to block the self.leftAutoCenterCameraButton widget to emit any signals when you set its state to checked.

Example:
https://github.com/Slicer/Slicer/blob/master/Libs/MRML/Widgets/qMRMLLinearTransformSlider.cxx#L203..L205

Comment (tavaughan) [inline LumpNav/LumpNav.py]: I see, so I'd want to block the signals when starting the function, then 'unblock' the signals after I've updated the GUI?

**Pull request merged**